package/MDAnalysis/lib/trans.pyx

```diff
@@ -0,0 +1,29 @@
+import numpy as np
+cimport cython
+cimport numpy as np
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def residue_positions(
+        float[:, :] coords not None,
```

Comment:
You can tell the compiler how the array is stored (Row or Column Major). That can be done by giving it the strides directly `float[:, ::1` is row major for example the C/Python standard. That should give a small boost but requires that the given arrays follow that memory layout.